core-java-modules/core-java-streams-maps/src/test/java/com/baeldung/streams/tomapvsgroupingby/ToMapVsGroupingByUnitTest.java

```diff
@@ -0,0 +1,203 @@
+package com.baeldung.streams.tomapvsgroupingby;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static com.google.common.collect.Sets.newHashSet;
+import static java.util.stream.Collectors.mapping;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Test;
+
+class City {
+
+    private String name;
+    private String country;
+
+    public City(String name, String country) {
+        this.name = name;
+        this.country = country;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getCountry() {
+        return country;
+    }
+
+    @Override
+    public final boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof City city)) {
+            return false;
+        }
+
+        return Objects.equals(name, city.name) && Objects.equals(country, city.country);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = Objects.hashCode(name);
+        result = 31 * result + Objects.hashCode(country);
+        return result;
+    }
+}
+
+public class ToMapVsGroupingByUnitTest {
+
+    private City paris = new City("Paris", "France");
+    private City berlin = new City("Berlin", "Germany");
+    private City london = new City("London", "UK");
+    private City newYork = new City("New York", "USA");
+    private City tokyo = new City("Tokyo", "Japan");
+
+    private City nice = new City("Nice", "France");
+    private City hamburg = new City("Hamburg", "Germany");
+    private City aachen = new City("Aachen", "Germany");
+
+    private City franceNull = new City(null, "France");
+    private City germanyNull = new City(null, "Germany");
+
+    @Test
+    void whenUsingToMap_thenGetExpectedResult() {
+        Map<String, City> result = Stream.of(paris, berlin, london, newYork, tokyo)
+            .collect(Collectors.toMap(City::getCountry, Function.identity()));
+
+        Map<String, City> expected = Map.of(
+            // @formatter:off
```

Comment:
What happens without `// @formatter:off`, does our auto formatter try to indent too much or something? It's a bit distracting having this syntax throughout the tests

Reply:
without "@formatter:off", all these parameters will be merged into one single line:

`Map<String, City> expected = Map.of("France", paris, "Germany", berlin, "UK", london, "USA", newYork, "Japan", tokyo);`

I think it's easier to read if we break the long line. 

BTW, those formatter-related comments are not in the article.

Reply:
Makes sense ok